tests/foreman/ui/test_host.py

```diff
@@ -513,6 +518,128 @@ def test_positive_delete(self):
+
+        @id: 8e61127c-d0a0-4a46-a3c6-22d3b2c5457c
+
+        @expectedresults: Only one specific host is returned by search
+
+        @CaseLevel: Integration
+        """
+        org = entities.Organization().create()
+        param_name = gen_string('alpha')
+        param_value = gen_string('alpha')
+        parameters = [{'name': param_name, 'value': param_value}]
+        param_host = entities.Host(
+            organization=org,
+            host_parameters_attributes=parameters,
+        ).create()
+        additional_host = entities.Host(organization=org).create()
+        with Session(self.browser) as session:
+            set_context(session, org=org.name)
+            # Check that hosts present in the system
+            for host in [param_host, additional_host]:
+                self.assertIsNotNone(self.hosts.search(host.name))
+            # Check that search by parameter returns only one host in the list
+            self.assertIsNotNone(
+                self.hosts.search(
+                    param_host.name,
+                    _raw_query='params.{0} = {1}'.format(
+                        param_name, param_value)
+                )
+            )
+            strategy, value = self.hosts._search_locator()
+            self.assertIsNone(self.hosts.wait_until_element(
+                (strategy, value % additional_host.name)))
+
+    @run_only_on('sat')
+    @tier2
+    def test_positive_search_by_parameter_with_prefix(self):
+        """Search by global parameter assigned to host using prefix 'not' and
+        any random string as parameter value to make sure that all hosts will
+        be present in the list
+
+        @id: a4affb90-1222-4d9a-94be-213f9e5be573
+
+        @expectedresults: All assigned hosts to organization are returned by
+        search
+
+        @CaseLevel: Integration
+        """
+        org = entities.Organization().create()
+        param_name = gen_string('alpha')
+        param_value = gen_string('alpha')
+        parameters = [{'name': param_name, 'value': param_value}]
+        param_host = entities.Host(
+            organization=org,
+            host_parameters_attributes=parameters,
+        ).create()
+        additional_host = entities.Host(organization=org).create()
+        with Session(self.browser) as session:
+            set_context(session, org=org.name)
+            # Check that hosts present in the system
+            for host in [param_host, additional_host]:
+                self.assertIsNotNone(self.hosts.search(host.name))
+            # Check that search by parameter with 'not' prefix returns both
+            # hosts in the list
+            self.assertIsNotNone(
+                self.hosts.search(
+                    param_host.name,
+                    _raw_query='not params.{0} = {1}'.format(
+                        param_name, gen_string('alphanumeric'))
+                )
+            )
+            strategy, value = self.hosts._search_locator()
```

Comment:
Why using protected method here? In case it is really needed, maybe would be better making it public, e.g., refactoring name from '_search_locator' to 'search_locator'